File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b)
    let cnt = 0
    let res = 0
    
    let i=0
    let j=people.length-1
    
    while(i <= j){
        if(people[i] + people[j] <= limit){
            // people = people.slice(i+1, j)
            i++
            j--
            cnt++
        } else {
            res++
            j--
        }
    }
    
    return cnt + res
    
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        
        int left=0, right=people.length-1;
        int cnt=0;
        while(left <= right){
            if(people[right] + people[left] <= limit){
                left++;
            }
            right--;
            cnt++;
        }
        
        return cnt;
    }
}